peercrawler.py

```diff
@@ -130,17 +133,23 @@ def handle_incoming(self, connection: socket.socket, address):
         while True:
             if time.time() - start_time > 60:
                 logger.debug(f"The time limit for receiving a keepalive has been exceeded for {address}, connection is now closing")
-                connection.close()
-                return
+                break
 
             header, payload = get_next_hdr_payload(connection)
-            if header.msg_type == message_type(message_type_enum.keepalive):
+            if header.msg_type == message_type(message_type_enum.telemetry_ack):
+                incoming_peer.telemetry = telemetry_req.telemetry_ack.parse(header, payload)
+                logger.debug(f"Received telemetry from {address}")
+
+            elif header.msg_type == message_type(message_type_enum.keepalive):
                 keepalive = message_keepalive.parse_payload(header, payload)
+                logger.debug(f"Received peers from {address}, connection is now closing")
                 self.add_peers(keepalive.peers)
 
-                logger.debug(f"Received peers from {address}, connection is now closing")
-                connection.close()
-                return
+                if incoming_peer.telemetry is not None:
```

Comment:
I think it would be better to check that both keepalive and telemetry are received.
And check one level higher, i.e. every time the loop is taken.
So that the code in whichever order the packets arrive.